src/diem/offchain/client.py

```diff
@@ -82,7 +125,27 @@ def send_request(
             raise CommandResponseError(cmd_resp)
         return cmd_resp
 
-    def process_inbound_request(self, request_sender_address: str, request_bytes: bytes) -> PaymentCommand:
+    def process_inbound_request(self, request_sender_address: str, request_bytes: bytes) -> Command:
+        """Validate and decode the `request_bytes` into `diem.offchain.command.Command` object.
+
+        Raises `diem.offchain.error.Error` with `protocol_error` when:
+
+        - `request_sender_address` is not provided.
+        - Cannot find on-chain account by the `request_sender_address`.
+        - Cannot get base url and public key from the on-chain account found by the `request_sender_address`.
+        - `request_bytes` is an invalid JWS message: JWS message format or content is invalid, or signature is invalid.
+        - Cannot decode `request_bytes` into `diem.offchain.types.command_types.CommandRequestObject`.
+        - Decoded `diem.offchain.types.command_types.CommandRequestObject` is invalid according to DIP-1 data structure requirements.
+
+        Raises `diem.offchain.error.Error` with `command_error` when `diem.offchain.types.command_types.CommandRequestObject.command` is `diem.offchain.payment_command.PaymentCommand`:
+
+        - `diem.offchain.types.payment_types.PaymentObject.sender` or `diem.offchain.types.payment_types.PaymentObject.sender`.address is invalid.
+        - `request_sender_address` is not sender or receiver actor address.
+        - For initial payment object, the `diem.offchain.types.payment_types.PaymentActionObject.amount` is under dual attestation limit (travel rule limit).
+        - For receiver actor statis is `ready_for_settlement`, recipient signature not set or it is invalid signature.
```

Comment:
what do you mean "it is invalid signature"?

Reply:
recipient signature is invalid signature, means when you use it verify the transaction metadata, the verification failed.